server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/nesteggcoin/NesteggcoinExtension.java

```diff
@@ -0,0 +1,121 @@
+/*************************************************************************************
+ * Copyright (C) 2014-2020 GENERAL BYTES s.r.o. All rights reserved.
+ *
+ * This software may be distributed and modified under the terms of the GNU
+ * General Public License version 2 (GPL2) as published by the Free Software
+ * Foundation and appearing in the file GPL2.TXT included in the packaging of
+ * this file. Please note that GPL2 Section 2[b] requires that all works based
+ * on this software must also be made publicly available under the terms of
+ * the GPL2 ("Copyleft").
+ *
+ * Contact information
+ * -------------------
+ *
+ * GENERAL BYTES s.r.o.
+ * Web      :  http://www.generalbytes.com
+ *
+ ************************************************************************************/
+package com.generalbytes.batm.server.extensions.extra.nesteggcoin;
+
+import com.generalbytes.batm.common.currencies.CryptoCurrency;
+import com.generalbytes.batm.common.currencies.FiatCurrency;
+import com.generalbytes.batm.server.extensions.*;
+import com.generalbytes.batm.server.extensions.FixPriceRateSource;
+import com.generalbytes.batm.server.extensions.extra.bitcoin.sources.coingecko.CoinGeckoRateSource;
+import com.generalbytes.batm.server.extensions.extra.dash.sources.coinmarketcap.CoinmarketcapRateSource;
+import com.generalbytes.batm.server.extensions.extra.nesteggcoin.wallets.nesteggd.NesteggdRPCWallet;
+import com.generalbytes.batm.server.extensions.extra.nesteggcoin.wallets.nesteggd.NesteggdUniqueAddressRPCWallet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.net.InetSocketAddress;
+import java.util.*;
+
+public class NesteggcoinExtension extends AbstractExtension{
+    private static final Logger log = LoggerFactory.getLogger(NesteggcoinExtension.class);
+
+    @Override
+    public String getName() {
+        return "BATM NestEggCoin extension";
+    }
+
+    @Override
+    public IWallet createWallet(String walletLogin, String tunnelPassword) {
+        try {
+        if (walletLogin !=null && !walletLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(walletLogin,":");
+            String walletType = st.nextToken();
+
+            if ("nesteggd".equalsIgnoreCase(walletType)
+                || "nesteggdnoforward".equalsIgnoreCase(walletType)) {
+                //"nesteggd:protocol:user:password:ip:port:accountname"
+
+                String protocol = st.nextToken();
+                String username = st.nextToken();
```

Comment:
Please stick to `user` since you've used it in parameters and even here on the line 52. Or even better if user is meant to be username please rename param values (at `batm-extensions.xml`) to username to provide customers better information what to fill in

Reply:
so just rename user to username?

Reply:
user is meant to be user for the rpcuser 
`//"nesteggd:protocol:user:password:ip:port:accountname"` would be matching the info you add to the conf file of the wallet daemon

Reply:
Yes please. It's better to use same names for the same value across the whole extension. And `username` is more descriptive than just `user` especially here where `username` is expected

Reply:
Ok, let me do that real quick

Reply:
You don't have to rush. Also please change it also in above mentioned `batm-extensions.xml`, you need to replace `<param name="user" />` by `<param name="username" />` for both cases - this is what will be displayed to the customer. Not some comment in the code.

Also this was not the main reason I requested changes -> please check my first comment - on the line 70 - value `"newyorkcoindnoforward"` is incorrect

Reply:
I missed the first comment, I have corrected those now

Reply:
OK. Now please let's make last change - revert commit `d132234b8531c8f73ce94310dd5d56c773ae72a3` you replaced `user` by `username` in `NewyorkcoinExtension` instead of this one (`NesteggcoinExtension`). So please make sure there wont't be any modification in `NewyorkcoinExtension` but here.

Reply:
ok 1 sec

Reply:
done, reverted my change to `NewyorkcoinExtension`

Reply:
> ok 1 sec

You really don't have to rush so much rather read carefully what people write to you. It saves time to both sides. You have made only half what you should have done. Reverting your change was just first step, you were also supposed to change it at correct class (`NesteggcoinExtension` - line 52). Actually that's something you tried to do in the first place but since you were in a such rush you made change to wrong file. And that's exactly what I'm talking about - slow down, read it carefully, think about solution and then !not earlier! make your changes. Otherwise you risk, that I or someone else won't have patience to constantly correct your mistakes and you'll have to wait longer time for any response.

You also don't have to make any changes, this PR is already acceptable.

Reply:
Ok so let me get this straight, I need to alter line 52 in `NesteggcoinExtension`? 
Oh I thought you meant I had to change user to username sorry for my misunderstanding

Reply:
Please don't change anything now. 

All this was simply about using the same term across this extension -> using `username` instead of `user` and `username` for the same value. Nothing else.

On the line 52 is comment ` //"nesteggd:protocol:user:password:ip:port:accountname"` it's just a comment, so there won't be any difference if you change it or not. We put comment like this one to the code because it represents value customer is going to set in CAS -> this makes it easier for reader to understand what is going to be parsed. Therefore ` //"nesteggd:protocol:username:password:ip:port:accountname"` is more correct.

Just to make sure -> these values are the ones you defined at `batm-extensions.xml` within each `<param name="XXX" />`

Reply:
ahh ok, I won't change anything, I just wanted to clarify  before I changed anything and thank you for the explanation